service/src/cl/iothub_registry_manager.ts

```diff
@@ -0,0 +1,968 @@
+  }
+
+  /**
+   * @method updateDeviceWithSas                              module: azure-iot-digitaltwins-service.IoTHubRegistryManager.updateDeviceWithSas
+   * @description                                             Updates a device identity on IoTHub using SAS authentication.
+   * @param {string}  deviceId                                The name (Id) of the device.
+   * @param {string}  eTag                                    The etag (if_match) value to use for the update operation.
+   * @param {string}  primaryKey                              Primary authentication key.
+   * @param {string}  secondaryKey                            Secondary authentication key.
+   * @param {boolean} isEnabled                               The status of the device. If the status disabled, a device cannot connect to the service.
+   * @returns Promise<DevicesCreateOrUpdateIdentityResponse>  The return object containing the updated device and the parsed HttpResponse.
+   * @memberof IoTHubRegistryManager
+   */
+  updateDeviceWithSas(
+    deviceId: string,
+    eTag: string,
+    primaryKey: string,
+    secondaryKey: string,
+    isEnabled: boolean
+  ): Promise<DevicesCreateOrUpdateIdentityResponse> {
+    const symmetricKey: SymmetricKey = {
+      primaryKey: primaryKey,
+      secondaryKey: secondaryKey
+    };
+
+    const authenticationMechanism: AuthenticationMechanism = {
+      type: 'sas',
+      symmetricKey: symmetricKey
+    };
+
+    const device: Device = {};
+    device.deviceId = deviceId;
+    device.etag = eTag;
+    if (isEnabled) {
+      device.status = 'enabled';
+    } else {
+      device.status = 'disabled';
+    }
+    device.authentication = authenticationMechanism;
+
+    const options: DevicesCreateOrUpdateIdentityOptionalParams = {
+      ifMatch: '*'
+    };
+
+    return this._pl.devices.createOrUpdateIdentity(deviceId, device, options);
+  }
+
+  /**
+   * @method updateDeviceWithX509                             module: azure-iot-digitaltwins-service.IoTHubRegistryManager.updateDeviceWithX509
+   * @description                                             Creates a device identity on IoTHub using X509 authentication.
+   * @param {string}  deviceId                                The name (Id) of the device.
+   * @param {string}  eTag                                    The etag (if_match) value to use for the update operation.
+   * @param {string}  primaryThumbprint                       Primary X509 thumbprint.
+   * @param {string}  secondaryThumbprint                     Secondary X509 thumbprint.
+   * @param {boolean} isEnabled                               The status of the device. If the status disabled, a device cannot connect to the service.
+   * @returns Promise<DevicesCreateOrUpdateIdentityResponse>  The return object containing the updated device and the parsed HttpResponse.
+   * @memberof IoTHubRegistryManager
+   */
+  updateDeviceWithX509(
```

Comment:
>updateDeviceWithX509 [](start = 2, length = 20)

Why is this called update?  There is not capability for update if there is no device object.

Reply:
This and others with update*

---
In reply to: [571791877](https://github.com/Azure/azure-iot-sdk-node/pull/923#discussion_r571791877) [](ancestors = 571791877)

Reply:
Fixed: updateDevice is introduced